### ISSUE TITLE
speaker support

### ISSUE THREAD

**Issue by jamesshao8** [jamesshao8/portapack-havoc-modified#11]
I enabled the speaker in software.
Its volume (together with mute or unmute) is controlled together with the headphone for the moment.
If anyone has a analog speaker can have a try. 
The port is a 3 pin port on the back of portapack, close to the battery 2 pin port.

This should be valid only for the original portapack h1, not for h2.

**Comment by michaelbuud19** [comment]
Have latest bin file to test 


Thanks

**Comment by jamesshao8** [comment]
[portapack-h1-havoc.zip](https://github.com/jamesshao8/portapack-havoc-modified/files/4566772/portapack-h1-havoc.zip)
Here's the bin. Remember to connect speaker to right and left side of the 3 pin port. Don't use the ground port in the middle. And the polarity of speaker matters, however if it's not correct, it won't damage the device, only causing it to reboot.
According to my test, when powered by usb, the maximum volume is 80. Larger than that will cause the portapack stop functioning.

**Comment by jamesshao8** [comment]
[portapack-h1-havoc.zip](https://github.com/jamesshao8/portapack-havoc-modified/files/4567967/portapack-h1-havoc.zip)
I added a switch on the top now, you can switch the speaker on of off to save battery power. green is on, grey is off.

**Comment by eried** [comment]
Why the brightness of the screen changes? 

At boot the icon is "grey" but audio is enabled anyways

**Comment by jamesshao8** [comment]
Are you using h2? This feature is mainly designed for h1 users as the speaker isn't enabled by default.
The brightness is changed due to the power consumption of speaker. if you use a 3.7v battery instead of usb power, it shouldn't change.
h2 users shouldn't enable the speaker, otherwise it will overheat the portapack board.

**Comment by eried** [comment]
Thanks, but there is not way to detect H2 from H1 from software right?

**Comment by RChadwick7** [comment]
Do you have any more details? I noticed the screen dimmed when turning up the volume, and the sound distorts badly at higher volume. I haven't looked closer yet, but my theory is that the audio amplifier is powered by the 3.3v LDO, but it likely would be better to power it directly from the LiPo.

**Comment by jamesshao8** [comment]
Yes, you need the 3.7v internal lipo battery to use the speaker properly. My experience is, if only using usb power, the volume cant be higher than 80. Remember to fully charge the lipo battery, otherwise the screen will also have problem.

If sound distorts, it might be because your speaker is saturated. My best volume is 90. If higher  volume is needed, you can try to stick the speaker to a metal case. The case will resonate with the speaker to make the sound louder.

**Comment by fabiov75** [comment]
> [portapack-h1-havoc.zip](https://github.com/jamesshao8/portapack-havoc-modified/files/4566772/portapack-h1-havoc.zip)
> Here's the bin. Remember to connect speaker to right and left side of the 3 pin port. Don't use the ground port in the middle. And the polarity of speaker matters, however if it's not correct, it won't damage the device, only causing it to reboot.
> According to my test, when powered by usb, the maximum volume is 80. Larger than that will cause the portapack stop functioning.

Hello: I'd like to understand why polarity speaker matters... I can't make mine working on H2... since it'8ohm speaker, isn't simmetrical?

thanks
